packages/block-editor/src/components/block-popover/index.js

```diff
@@ -47,22 +47,48 @@ function BlockPopover(
 		};
 	}, [ selectedElement, lastSelectedElement, __unstableRefreshSize ] );
 
+	const popoverAnchor = useMemo(
+		() => ( {
+			getBoundingClientRect() {
+				const selectedBCR = selectedElement.getBoundingClientRect();
+				const lastSelectedBCR =
+					lastSelectedElement.getBoundingClientRect();
+
+				// Get the biggest rectangle that encompasses completely the currently
+				// selected element and the last selected element:
+				// - for top/left coordinates, use the smaller numbers
+				// - for the bottom/right coordinates, use the larget numbers
+				const left = Math.min( selectedBCR.left, lastSelectedBCR.left );
```

Comment:
This logic is an improved version of [the same logic inside the `Popover` component](https://github.com/WordPress/gutenberg/blob/4e7508b74e08f03f61b56ecafa6d57fdfedfb3e7/packages/components/src/popover/utils.js)